packages/vkui/src/components/CalendarHeader/CalendarHeader.tsx

```diff
@@ -92,19 +96,27 @@ export const CalendarHeader = ({
     [onChange, viewDate],
   );
 
+  const currentYear = viewDate.getFullYear();
+  const currentMonth = viewDate.getMonth();
+
   const months = React.useMemo(
     () =>
       getMonths(locale).map(({ value, label }) => ({
         value,
         label: <span className={styles['CalendarHeader__month']}>{label}</span>,
+        disabled: isMonthDisabled && isMonthDisabled(value),
       })),
-    [locale],
+    [locale, currentYear, isMonthDisabled, isYearDisabled],
```

Comment:
```suggestion
    [locale, isMonthDisabled],
```

Вроде бы лишние зависимости. Или это намеренно?

Reply:
https://github.com/VKCOM/VKUI/pull/6937/commits/8d73f12ca4b6d8601d405ae9fc6b602b9fc7a351